app/models/apartment.rb

```diff
@@ -0,0 +1,18 @@
+class Apartment < ApplicationRecord
+  belongs_to :owner, class_name: 'User'
+  has_many :rentals, dependent: :destroy
+  has_many :markets, dependent: :destroy
+  has_many :user_apartments, dependent: :destroy
```

Comment:
please why you are using many to many relationships, it would be better to use 
```RUBY
has_many :user, through: user_apartements
```

- [ ] [visit this page to know more about has_many through](https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association)

Reply:
Hello, @youmari, we have refactored the models and association as per the suggestions offered. Please look at the new changes